osquery/tables/system/darwin/firewall.cpp

```diff
@@ -43,6 +44,61 @@ const std::map<std::string, std::string> kTopLevelStringKeys{
     {"version", "version"},
 };
 
+Status parseApplicationAliasData(const std::string& data, std::string& result) {
+
+  std::string decoded_data = base64Decode(data);
+  if (decoded_data.length() == 0) {
+    return Status(1, "Failed to base64 decode data");
+  }
+
+  CFDataRef resourceData = CFDataCreate(nullptr, (const UInt8*)decoded_data.data(), decoded_data.length());
```

Comment:
Can you `static_cast<const Uint8*>(decoded_data.data())`

Reply:
It doesn't appear that I can cast that way.  Is there a different approach I can take?

> osquery/tables/system/darwin/firewall.cpp:55:50: error: static_cast from 'const value_type *' (aka 'const char *') to 'const UInt8 *' (aka 'const unsigned char *') is not allowed
>  CFDataRef resourceData = CFDataCreate(nullptr, static_cast<const UInt8*>(decoded_data.data()), decoded_data.length());

Reply:
static_cast is used for standard conversion, it cant convert one type of pointer to another type of pointer.
Here your options are -
1 . static_cast<const UInt8*>(static_cast<const void*>((decoded.c_str())))
2 . reinterpret_cast<const UInt8*>(decoded.c_str())

The message you pasted is kind of misleading. Have you hand edited the message ?
Reason -
"'const UInt8 *' (aka 'const unsigned char ') is not allowed" should be "'const UInt8 *' (aka 'const unsigned char *') is not allowed"
OR
Since you mentioned in the code "static_cast<const UInt8>(decoded_data.data())" 
so message should have been "'const UInt8 ' (aka 'const unsigned char ') is not allowed" instead of "'const UInt8 *' (aka 'const unsigned char ') is not allowed"
OR
There is a bug in the compiler diagnostic message.

Reply:
Thanks for the recommendations.  I'm using option #1 you presented above because it sounds a little safer from what I've been reading.

I checked the error message again and the markdown block style was wiping away the last asterisk on the error message.  The correct error is:

`osquery/osquery/tables/system/darwin/firewall.cpp:56:16: error: static_cast from 'const value_type *' (aka 'const char *') to 'const UInt8 *' (aka 'const unsigned char *') is not allowed`
`      nullptr, static_cast<const UInt8*>(decoded_data.data()), decoded_data.length());`